### ISSUE TITLE
ERROR Error: EACCES: permission denied, open '/usr/share/kibana/config/kibana.yml

### ISSUE THREAD

**Issue by ervikrant06** [deviantony/docker-elk#194]
Hi, I am getting this error while trying to bring the ELK stackup. All the containers are in exited status. Can you please provide some guidance on it?

```console
# docker-compose up

Creating network "dockerelk_elk" with driver "bridge"
Building elasticsearch
[...]
Successfully built 865b21b970de
WARNING: Image for service elasticsearch was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Building logstash
[...]
Successfully built 99db48525fb9
WARNING: Image for service logstash was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Building kibana
[...]
Successfully built c8dca8239b98
WARNING: Image for service kibana was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating dockerelk_elasticsearch_1 ...
Creating dockerelk_elasticsearch_1 ... done
Creating dockerelk_logstash_1 ...
Creating dockerelk_kibana_1 ...
Creating dockerelk_logstash_1
Creating dockerelk_logstash_1 ... done
Attaching to dockerelk_elasticsearch_1, dockerelk_kibana_1, dockerelk_logstash_1
elasticsearch_1  | Exception in thread "main" 2017-11-03 10:46:11,222 main ERROR No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
elasticsearch_1  | SettingsException[Failed to load settings from /usr/share/elasticsearch/config/elasticsearch.yml]; nested: AccessDeniedException[/usr/share/elasticsearch/config/elasticsearch.yml];
elasticsearch_1  |      at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:102)
elasticsearch_1  |      at org.elasticsearch.cli.EnvironmentAwareCommand.createEnv(EnvironmentAwareCommand.java:75)
elasticsearch_1  |      at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:70)
elasticsearch_1  |      at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134)
elasticsearch_1  |      at org.elasticsearch.cli.Command.main(Command.java:90)
elasticsearch_1  |      at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91)
elasticsearch_1  |      at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84)
elasticsearch_1  | Caused by: java.nio.file.AccessDeniedException: /usr/share/elasticsearch/config/elasticsearch.yml
elasticsearch_1  |      at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
elasticsearch_1  |      at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
elasticsearch_1  |      at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
elasticsearch_1  |      at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
elasticsearch_1  |      at java.nio.file.Files.newByteChannel(Files.java:361)
elasticsearch_1  |      at java.nio.file.Files.newByteChannel(Files.java:407)
elasticsearch_1  |      at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:384)
elasticsearch_1  |      at java.nio.file.Files.newInputStream(Files.java:152)
elasticsearch_1  |      at org.elasticsearch.common.settings.Settings$Builder.loadFromPath(Settings.java:1032)
elasticsearch_1  |      at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:100)
elasticsearch_1  |      ... 6 more
logstash_1       | 2017/11/03 10:46:11 error: open /usr/share/logstash/config/logstash.yml: permission denied
dockerelk_elasticsearch_1 exited with code 1
dockerelk_logstash_1 exited with code 1
kibana_1         | FATAL CLI ERROR Error: EACCES: permission denied, open '/usr/share/kibana/config/kibana.yml'
kibana_1         |     at Error (native)
kibana_1         |     at Object.fs.openSync (fs.js:641:18)
kibana_1         |     at fs.readFileSync (fs.js:509:33)
kibana_1         |     at files.map.path (/usr/share/kibana/src/cli/serve/read_yaml_config.js:10:78)
kibana_1         |     at Array.map (native)
kibana_1         |     at exports.default (/usr/share/kibana/src/cli/serve/read_yaml_config.js:10:23)
kibana_1         |     at readServerSettings (/usr/share/kibana/src/cli/serve/serve.js:48:51)
kibana_1         |     at getCurrentSettings (/usr/share/kibana/src/cli/serve/serve.js:110:16)
kibana_1         |     at Command.<anonymous> (/usr/share/kibana/src/cli/serve/serve.js:112:24)
kibana_1         |     at next (native)
dockerelk_kibana_1 exited with code 1
```

**Comment by antoineco** [comment]
> AccessDeniedException[/usr/share/elasticsearch/config/elasticsearch.yml];

This is the issue, please check the note in this section in order to solve permission errors:
https://github.com/deviantony/docker-elk#how-can-i-persist-elasticsearch-data

**Comment by ervikrant06** [comment]
I tried to give the permissions using your command but still hitting the same issue. Even giving full permissions doesn't help. 

~~~
[root@centos-monitoring docker-elk]# chmod -v ugo+rwx     elasticsearch/config/*.yml     kibana/config/*.yml     logstash/*/*.{yml,conf}
mode of ‘elasticsearch/config/elasticsearch.yml’ changed from 0647 (rw-r--rwx) to 0777 (rwxrwxrwx)
mode of ‘kibana/config/kibana.yml’ changed from 0647 (rw-r--rwx) to 0777 (rwxrwxrwx)
mode of ‘logstash/config/logstash.yml’ changed from 0647 (rw-r--rwx) to 0777 (rwxrwxrwx)
mode of ‘logstash/pipeline/logstash.conf’ changed from 0647 (rw-r--rwx) to 0777 (rwxrwxrwx)
~~~

Still it end up with same error. Am I missing something?

~~~
[root@centos-monitoring docker-elk]# docker-compose up
Starting dockerelk_elasticsearch_1 ...
Starting dockerelk_elasticsearch_1 ... done
Starting dockerelk_logstash_1 ...
Starting dockerelk_kibana_1 ...
Starting dockerelk_kibana_1
Starting dockerelk_logstash_1 ... done
Attaching to dockerelk_elasticsearch_1, dockerelk_kibana_1, dockerelk_logstash_1
logstash_1       | 2017/11/05 04:10:07 error: open /usr/share/logstash/config/logstash.yml: permission denied
dockerelk_logstash_1 exited with code 1
elasticsearch_1  | Exception in thread "main" 2017-11-05 04:10:07,996 main ERROR No log4j2 configuration file found. Using default configuration: logging only errors to the console. Set system property 'log4j2.debug' to show Log4j2 internal initialization logging.
elasticsearch_1  | SettingsException[Failed to load settings from /usr/share/elasticsearch/config/elasticsearch.yml]; nested: AccessDeniedException[/usr/share/elasticsearch/config/elasticsearch.yml];
elasticsearch_1  |      at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:102)
elasticsearch_1  |      at org.elasticsearch.cli.EnvironmentAwareCommand.createEnv(EnvironmentAwareCommand.java:75)
elasticsearch_1  |      at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:70)
elasticsearch_1  |      at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134)
elasticsearch_1  |      at org.elasticsearch.cli.Command.main(Command.java:90)
elasticsearch_1  |      at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91)
elasticsearch_1  |      at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84)
elasticsearch_1  | Caused by: java.nio.file.AccessDeniedException: /usr/share/elasticsearch/config/elasticsearch.yml
elasticsearch_1  |      at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
elasticsearch_1  |      at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
elasticsearch_1  |      at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
elasticsearch_1  |      at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
elasticsearch_1  |      at java.nio.file.Files.newByteChannel(Files.java:361)
elasticsearch_1  |      at java.nio.file.Files.newByteChannel(Files.java:407)
elasticsearch_1  |      at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:384)
elasticsearch_1  |      at java.nio.file.Files.newInputStream(Files.java:152)
elasticsearch_1  |      at org.elasticsearch.common.settings.Settings$Builder.loadFromPath(Settings.java:1032)
elasticsearch_1  |      at org.elasticsearch.node.InternalSettingsPreparer.prepareEnvironment(InternalSettingsPreparer.java:100)
elasticsearch_1  |      ... 6 more
dockerelk_elasticsearch_1 exited with code 1
kibana_1         | FATAL CLI ERROR Error: EACCES: permission denied, open '/usr/share/kibana/config/kibana.yml'
kibana_1         |     at Error (native)
kibana_1         |     at Object.fs.openSync (fs.js:641:18)
kibana_1         |     at fs.readFileSync (fs.js:509:33)
kibana_1         |     at files.map.path (/usr/share/kibana/src/cli/serve/read_yaml_config.js:10:78)
kibana_1         |     at Array.map (native)
kibana_1         |     at exports.default (/usr/share/kibana/src/cli/serve/read_yaml_config.js:10:23)
kibana_1         |     at readServerSettings (/usr/share/kibana/src/cli/serve/serve.js:48:51)
kibana_1         |     at getCurrentSettings (/usr/share/kibana/src/cli/serve/serve.js:110:16)
kibana_1         |     at Command.<anonymous> (/usr/share/kibana/src/cli/serve/serve.js:112:24)
kibana_1         |     at next (native)
dockerelk_kibana_1 exited with code 1
~~~

**Comment by antoineco** [comment]
Check if you have SELinux enabled, there's also a note about it in the README.

**Comment by metju312** [comment]
I already done: "docker-compose up -d" and I had the same error.
Then I set my "SELINUX=permissive" value in /etc/selinux/config file
And after that: "docker-compose down" / "docker-compose up -d"
Have the same error.

**Comment by antoineco** [comment]
> Then I set my "SELINUX=permissive" value in /etc/selinux/config 

@metju312 this will disable SELinux only after the next boot. `sudo setenforce 0` is what you're looking for. Or better: follow the instructions in the README.

@ervikrant06 is the issue solved for you?

**Comment by metju312** [comment]
# setenforce 0   and it works.

**Comment by antoineco** [comment]
It's a very bad thing to do from a security perspective though :) Please read the first section of the README carefully, it provides a better solution.

**Comment by metju312** [comment]
Right thanks :)

**Comment by metju312** [comment]
But I don't understand section "SELinux" of README :/

**Comment by ervikrant06** [comment]
Thanks. it worked for me after changing selinux settings as mentioned in README.
